experimental/Schemes/CoherentSheaves.jl

```diff
@@ -622,7 +987,199 @@ function _pushforward(f::Hecke.Map{<:Ring, <:Ring}, I::Ideal, M::SubQuo)
   return MR, ident
 end
 
-function is_locally_free(M::AbsCoherentSheaf)
-  return all(U->is_projective(M(U)), affine_charts(scheme(M)))
+@attr Bool function is_locally_free(M::AbsCoherentSheaf)
+  return all(U->is_projective(M(U))[1], affine_charts(scheme(M)))
+end
+
+#@attr Covering function trivializing_covering(M::AbsCoherentSheaf)
+@attr function trivializing_covering(M::AbsCoherentSheaf)
+  X = scheme(M)
+  OOX = OO(X)
+  patch_list = Vector{AbsSpec}()
+  for U in affine_charts(X)
+    patch_list = vcat(patch_list, _trivializing_covering(M, U))
+  end
+  C = Covering(patch_list)
+  fill_with_lazy_glueings!(C, X)
+  return C
+end
+
+@attr function trivializing_covering(M::HomSheaf)
+  X = scheme(M)
+  OOX = OO(X)
+  # The problem is that every module of a HomSheaf must know that it is 
+  # a hom-module. Hence, the way to go is to pass through a common 
+  # refinement of domain and codomain and recreate all the hom modules 
+  # as free modules on this covering. 
+  #
+  # But for this, it is not yet clear where to locate the patches of these 
+  # refinements in the tree and how to deal with the restriction maps in a 
+  # clean way. Say M = Hom(F, G) where F is trivialized on {Uᵢ} and G on 
+  # {Vⱼ}. Then W = Uᵢ∩ Vⱼ would have to be a PrincipalOpenSubset of both 
+  # Uᵢ and Vⱼ for the restrictions of F and G to induce the proper job 
+  # on restrictions to M(W) automatically. 
+  # Hence, we need to manually prescribe how to trivialize and restrict 
+  # M on the Ws.
+  dom_triv = trivializing_covering(domain(M))
+  cod_triv = trivializing_covering(codomain(M))
+  patch_list = AbsSpec[]
+  for U in patches(dom_triv)
+    for V in patches(cod_triv)
+      success, W = _have_common_ancestor(U, V)
+      if success
+        incU = _flatten_open_subscheme(U, W)
+        incV = _flatten_open_subscheme(V, W)
+        UV = intersect(codomain(incU), codomain(incV))::PrincipalOpenSubset
+        push!(patch_list, UV)
+
+        dom_UV, dom_res = change_base_ring(OOX(U, UV), domain(M)(U))
+        add_incoming_restriction!(domain(M), U, dom_UV, dom_res)
+        add_incoming_restriction!(domain(M), W, dom_UV, 
+                                  compose(domain(M)(W, U), dom_res))
+        object_cache(domain(M))[UV] = dom_UV
+
+        cod_UV, cod_res = change_base_ring(OOX(V, UV), codomain(M)(V))
+        add_incoming_restriction!(codomain(M), V, cod_UV, cod_res)
+        add_incoming_restriction!(codomain(M), W, dom_UV, 
+                                  compose(codomain(M)(W, V), cod_res))
+        object_cache(codomain(M))[UV] = cod_UV
```

Comment:
@simonbrandhorst : This is another instance of where one needs to manually construct restrictions and, hence, also needs to take care of the caching.